### ISSUE TITLE
Documentation: how to use snapper with btrfs incremental send / receive?

### ISSUE THREAD

**Issue by lkraav** [openSUSE/snapper#148]
I for one can't figure it out. Created snapshots seem to be have no parent subvol to increment against. What is the process here?

    $ sudo btrfs send -p /.snapshots/630/snapshot /.snapshots/893/snapshot | sudo btrfs receive /mnt/backup
    At subvol /.snapshots/630/snapshot
    At snapshot snapshot
    ERROR: could not find parent subvolume

**Comment by aschnell** [comment]
Is the error on the sending or the receiving side?

I have done some tests and have it working for me. The setup is more complicated that explained in the btrfs wiki since .snapshots itself is a subvolume. So you have to create the directories yourself (or somehow also make snapshots of .snapshots, esp. if you also want snappers metadata for the snapshots to be synced).

bootstrap
```
btrfs subvolume create /backup/.snapshots
mkdir /backup/.snapshots/1
btrfs send /btrfs/.snapshots/1/snapshot | btrfs receive /backup/.snapshots/1
```

sync diff from 1 to 2
```
mkdir /backup/.snapshots/2
btrfs send -p /btrfs/.snapshots/1/snapshot /btrfs/.snapshots/2/snapshot | btrfs receive /backup/.snapshots/2
```

Does that help?

**Comment by Eisfreak7** [comment]
Would it be possible to provide some sort of post-snapshot hook or native integration of btrfs send? I'd like to push a copy of my .snapshots folder to a server with a btrfs-formatted hdd (similar to what I've done with 'backintime' before, only more efficient).

**Comment by aschnell** [comment]
Hook are handled in https://github.com/openSUSE/snapper/issues/28. Unfortunately nobody had time to implement them yet.

**Comment by aschnell** [comment]
Yes, meanwhile the tool ```snbk``` was added: https://snapper.io/2024/11/11/snbk.html
